tests/ref/html/basic-table.html

```diff
@@ -0,0 +1,10 @@
+<!DOCTYPE html>
+<html>
+  <head>
+    <meta charset="utf-8">
+    <meta name="viewport" content="width=device-width, initial-scale=1">
+  </head>
+  <body>
+    <table><thead><tr><td>The</td><td>first</td><td>and</td></tr><tr><td>the</td><td>second</td><td>row</td></tr></thead><tbody><tr><td>Foo</td><td rowspan="2">Baz</td><td>Bar</td></tr><tr><td>1</td><td>2</td></tr><tr><td colspan="2">3</td><td>4</td></tr></tbody><tfoot><tr><td>The</td><td>last</td><td>row</td></tr></tfoot></table>
```

Comment:
Tables can be pretty-printed:

According to spec section 4.9.12.1, no. 8, elements that are not `colgroup`, `thead`, `tbody`, `tfoot`, or `tr` as well as the first `caption` are skipped within a table (particularly whitespace text nodes). In row groups (`thead`, `tbody`, `tfoot`) only `tr` elements are processed and in rows, only `td` and `th` are processed. I'd keep the strict mode for `td` and `th`.

Reply:
What do you think about the following?

~~~ html
    <table>
      <thead>
        <tr>
          <td>The</td><td>first</td><td>and</td>
        </tr>
        <tr>
          <td>the</td><td>second</td><td>row</td>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>Foo</td><td rowspan="2">Baz</td><td>Bar</td>
        </tr>
        <tr>
          <td>1</td><td>2</td>
        </tr>
        <tr>
          <td colspan="2">3</td><td>4</td>
        </tr>
      </tbody>
      <tfoot>
        <tr>
          <td>The</td><td>last</td><td>row</td>
        </tr>
      </tfoot>
    </table>
~~~

Reply:
Spaces between `td`s are allowed as well, though these cells are short enough to look good without spaces

Reply:
I left out the spaces between `td`s for now, as I find the output much more readable that way.